DesktopEdge/MainWindow.xaml.cs

```diff
@@ -792,53 +792,55 @@ private void MonitorClient_OnShutdownEvent(object sender, StatusEvent e)
 		}
 
 		private void MonitorClient_OnServiceStatusEvent(object sender, MonitorServiceStatusEvent evt) {
-			try {
-				if (evt.Message?.ToLower() == "upgrading") {
-					logger.Info("The monitor has indicated an upgrade is in progress. Shutting down the UI");
-					notifyIcon.Visible = false;
-					notifyIcon.Icon.Dispose();
-					notifyIcon.Dispose();
-					Application.Current.Shutdown();
-				}
-				state.AutomaticUpdatesEnabledFromString(evt.AutomaticUpgradeDisabled);
-				MainMenu.ShowUpdateAvailable();
-				logger.Debug("MonitorClient_OnServiceStatusEvent: {0}", evt.Status);
-				Application.Current.Properties["ReleaseStream"] = evt.ReleaseStream;
+			this.Dispatcher.Invoke(() => {

```

Comment:
this block here should fix the error with showing the notification:
```
[2023-10-24T10:22:23.036Z]  WARN	ZitiDesktopEdge.MainWindow	unexpected exception in MonitorClient_OnShutdownEvent? The calling thread cannot access this object because a different thread owns it.	System.InvalidOperationException: The calling thread cannot access this object because a different thread owns it.
   at System.Windows.Threading.Dispatcher.VerifyAccess()
   at System.Windows.DependencyObject.SetValue(DependencyProperty dp, Object value)
   at ZitiDesktopEdge.MainMenu.ShowUpdateAvailable()
   at ZitiDesktopEdge.MainWindow.MonitorClient_OnServiceStatusEvent(Object sender, MonitorServiceStatusEvent evt)
```
it's all the same code, just indented